src/main/kotlin/fi/aalto/cs/apluscourses/ui/Utils.kt

```diff
@@ -19,13 +28,51 @@ object Utils {
         }
     }
 
-    fun Row.myLink(text: String, icon: Icon, action: (ActionEvent) -> Unit) =
+    fun Row.myLink(text: String, icon: Icon, action: (ActionEvent) -> Unit): Cell<ActionLink> =
         link(text, action).applyToComponent {
             setIcon(icon, false)
         }
 
-    fun Row.myActionLink(text: String, icon: Icon, action: AnAction) =
+    fun Row.myActionLink(text: String, icon: Icon, action: AnAction): Cell<AnActionLink> =
         cell(AnActionLink(text, action)).applyToComponent {
             setIcon(icon, false)
         }
+
+    /**
+     * A wrapper for JBList to be used with the UI DSL.
+     */
+    inline fun <reified T> Row.list(
+        items: ObservableMutableProperty<List<T>>,
+        renderer: ListCellRenderer<in T?>
+    ): Cell<JBList<T>> {
```

Comment:
I really like this function and the next one, too. Do you know if Observable*s link via weak references so that the UI elements can be GC'ed even if the model objects still exist? Or is the lifecycle of models similar to the corresponding UI elements?

Reply:
I think it's tied to the lifecycle of the UI element but not 100% sure